app/uk/gov/hmrc/lisaapi/controllers/ReinstateAccountController.scala

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.lisaapi.controllers
+
+import play.api.Logger
+import play.api.libs.json.{JsObject, JsPath, Json}
+import play.api.mvc.{Action, AnyContent}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.lisaapi.LisaConstants
+import uk.gov.hmrc.lisaapi.metrics.{LisaMetricKeys, LisaMetrics}
+import uk.gov.hmrc.lisaapi.models._
+import uk.gov.hmrc.lisaapi.services.{AuditService, ReinstateAccountService}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class ReinstateAccountController extends LisaController with LisaConstants {
+
+  val service: ReinstateAccountService = ReinstateAccountService
+  val auditService: AuditService = AuditService
+
+
+
+  def reinstateAccount (lisaManager: String, accountId: String): Action[AnyContent] = Action.async{ implicit request =>
+    implicit val startTime = System.currentTimeMillis()
+    LisaMetrics.startMetrics(startTime, LisaMetricKeys.REINSTATE)
+    withValidLMRN(lisaManager) {
+      withValidAccountId(accountId) {
+        processReinstateAccount(lisaManager, accountId)
+      }
+    }
+  }
+
+
+
+  private def processReinstateAccount(lisaManager: String, accountId: String)(implicit hc: HeaderCarrier, startTime:Long) = {
+    service.reinstateAccountService(lisaManager, accountId).map { result =>
+      result match {
+        case ReinstateLisaAccountSuccessResponse => {
+          LisaMetrics.incrementMetrics(startTime, LisaMetricKeys.REINSTATE)
+
+          auditService.audit(
+            auditType = "accountReinstated",
+            path = getReinstateEndpointUrl(lisaManager, accountId),
+            auditData = Map(ZREF -> lisaManager, "accountId" -> accountId)
+          )
+
+          val data = ApiResponseData(message = "The account has been re-instated to a status of open", accountId = Some(accountId))
```

Comment:
We should be keeping all the API responses together in one place (currently the Responses.scala file)